.github/workflows/gradle-build.yml

```diff
@@ -0,0 +1,30 @@
+# This workflow will build a Java project with Gradle and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Java CI with Gradle
+
+on:
+  push:
+    branches: [ dev, master ]
+  pull_request:
+    branches: [ dev, master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Easy detect-secrets
+      uses: RobertFischer/detect-secrets-action@v2.0.0
+    - name: Set up JDK 11
+      uses: actions/setup-java@v2
+      with:
+        java-version: '11'
+        distribution: 'adopt'
+        cache: gradle
+    - name: Grant execute permission for gradlew
+      run: chmod +x gradlew
+    - name: Build with Gradle
+      run: ./gradlew build
```

Comment:
deployment stages
https://github.com/microsoftgraph/msgraph-sdk-java-core/blob/54ae401d342a07eef337086f63cca1da5a79d59c/.azure-pipelines/buildAndPackage.yml#L51

Reply:
should upload the artifacts (jars mostly) no need to upload the test results, sonarcloud already does that.
And don't upload the secure files, we can always checkout the repo and re-download them (as well as jars) in the next stages.